### PR TITLE
Small changes to mobs on fire

### PR DESCRIPTION
**Pull request by Ergovisavi**
- Humans heat up faster based on fire stacks
- Increased the amount of damage you get from being on fire at 480+ bodytemp (8->10)
- Decreased the amount of fire stacks lost over time
- Increased the amount of fire stacks gained from hotspots

### CODE REVIEW

**Comment by PKPenguin321** [comment]
yes :+1:

**Comment by Firecage** [comment]
I disagree to not losing firestacks over time.

**Comment by Miauw62** [comment]
lmao you're wrong firecage
if you want to get rid of your firestacks just run into other people.

**Comment by phil235** [comment]
Being permanently on fire because you were splashed with 1 unit of fuel is unnecessary. I'd be okay with making the fire going out more slowly but not remove that feature entirely.

**Comment by RemieRichards** [comment]
I'm happy with everything here except the confusion. There's no skill to surviving if you're that on fire, no desperate attempt to get to a fire extinguisher, just random walking and death

:+1: to everything but confusion
:-1: to confusion

**Comment by phil235** [comment]
Do we really want dead bodies on the ground to be indefinitely on fire and turning people on fire when crossing them?

**Comment by Nanotransistor** [comment]
@phil235 YES

**Comment by KorPhaeron** [comment]
@phil235 

Yes

It shouldn't cause confusion though.

**Comment by Perakp** [comment]
Bodies on fire forever is bad.

**Comment by Miauw62** [comment]
yeah, it should consume firestacks, just slowly.
they should never go away if you're not on fire though (unless you take a shower i guess?)

**Comment by phil235** [comment]
@Miauw62 It's already the case, the fire_stacks don't go away when not on fire.

**Comment by Ergovisavi** [comment]
@phil235 
>Being permanently on fire because you were splashed with 1 unit of fuel is unnecessary.

This is a requirement, or else the entire system falls flat on its face. If you're not forced to take action due to being on fire, the system has no point in existing. All players have access to multiple methods of shedding fire stacks and extinguishing themselves, unless they're stunned. And if you're stunned, well, you have other problems. You must EVENTUALLY take action.

Also, as it seems some people are opposed to the confusion aspect, I will seek some other method of making it more dangerous without simply making it faster killing.

**Comment by phil235** [comment]
@Ergovisavi I think you're overestimating the speed at which the fire_stack disappears. Right now if you smash a bottle of alcohol on someone's head, it will take more than 30 seconds to go out on its own. And if you lower the speed from 0.2 to 0.1 then the alcohol bottle will take more than a minute. Compare it to resisting the fire which removes 5 stacks, meaning you need to wait 25 life cycles to have the same effect as resisting (and it could 50 if you lower the speed).

**Comment by Lumipharon** [comment]
Fire is super shit right now, but infinite burning (aka corpse on the ground setting everyone on fire 5evah) was even more awful.

**Comment by PKPenguin321** [comment]
i dunno man, strategic placement of burning corpses underneath airlocks to ignite incoming attackers is pretty hilarious

maybe make fire not permanent until you have a certain number of fire stacks/get ignited by a flamethrower? idk

i really like this PR though

**Comment by Lumipharon** [comment]
I would rather the old system, where firestacks decreased overtime, but you wouldn't lose your last stack until you got rid of it manually, ON CONDITION that when you're down to your last stack, you can't spread it to others, which was just really dumb, since spreading is supposed to split the stacks between the people, and when there is only one stack to split, it basicaly just multiplied it instead.

**Comment by Ergovisavi** [comment]
@Lumipharon 

The problem with that system is that it's unnecessarily confusing due to lack of visual difference between the two different states, all we have sprites for is "on fire" and "not on fire", informing players that they're in a state of "on fire but can't trade stacks" or "on fire and can't trade stacks" is very confusing, add in the fact that fire stacks aren't communicated and we have a super weirdass mechanic. The point is to keep it simple.

**Comment by Firecage** [comment]
I still say it is a bad idea now to make firestacks reduce, since corpses or people being permanently on fire is shit, especially if its from 1 unit of welding fuel.

**Comment by Ergovisavi** [comment]
> since corpses or people being permanently on fire is shit

@Firecage And why do you believe this to be the case?

**Comment by RemieRichards** [comment]
Update; Confusion is still ass, remove.

**Comment by Ergovisavi** [comment]
@RemieRichards I'll get around to it either tonight or tomorrow, think I'll just crank up how quickly fire stacks affect your temperature even further instead of the confusion.

**Comment by RemieRichards** [comment]
@Ergovisavi that would be acceptable.

**Comment by Ergovisavi** [comment]
Scrapped the confusion aspect. Increased the rate that fire_stacks heat you up by, which, amusingly, does not really effect how fast it kills significantly.

Time to crit is 28 seconds at a full 20 fire stacks, but you start taking damage much sooner. As for concerns of eternally-on-fire-bodies becoming an issue... the station is surrounded in fire extinguishers.

**Comment by KorPhaeron** [comment]
30 seconds is literal years in combat. Enough time to stun and beat the other guy to death before casually putting yourself out.

On second thought the confusion as a penalty for ignoring fire might be a good idea after all.

**Comment by Ergovisavi** [comment]
30 seconds is a long time in combat, and definitely more than enough time for someone to stun and beat you to death. But think about it, is that really the standard you want to hold things to? The insta-stun lol I tase and remove you from round in seconds gg nonsense we have?

I certainly don't want being on fire to be like that. It's intended to be a dangerous pseudo-stun (And source of visual entertainment when people freak out in the middle of escape while on fire and it spreads) and I think this will put it back where it should be.

Worst case scenario I just slowly nudge fire up to be more and more lethal over time, but right now it's good to just nail the baseline.

**Comment by phil235** [comment]
Can we at least try to make fire consumption slower instead of going straight to removal?

**Comment by Ergovisavi** [comment]
Okay, the problem with simply lowering the consumption rate is that it makes flamethrowers/fiery explosions etc useless. Standing in a hotspot only gives you 0.5 fire stacks, although this increases for every hotspot you enter. 

So the two options are increasing the amount of fire stacks you gain on entering a hotspot, which may make flamethrowers or just running through plain 'ol fire very deadly, or we can simply adjust the fire stack consumption to the point where it would take ten years for it to disperse, but eventually would.

**Comment by duncathan** [comment]
I like the idea of buffing hotspots. If you're running through fire, you're gonna be very on fire.

**Comment by Miauw62** [comment]
I've also heard people asking for fire to last longer, which would also be cool.

**Comment by phil235** [comment]
Yeah, 0.5 fire_stack for crossing a hotspot sounds ridiculously low.

**Comment by bgobandit** [comment]
if corpses are on fire for a certain amount of time they should turn into remains (ie unclonable) and the fire should go out

**Comment by PKPenguin321** [comment]
>if corpses are on fire for a certain amount of time they should turn into remains (ie unclonable) and the fire should go out

too much of a balance change to be in the scope of this PR

**Comment by Ergovisavi** [comment]
I'll get back to this when swarmers are merged, in the meantime [DO NOT MERGE] please

**Comment by PKPenguin321** [comment]
swarmers are now merged

**Comment by Razharas** [comment]
@Ergovisavi

**Comment by Ergovisavi** [comment]
Yeah yeah I'll get on this when I get home

**Comment by Ergovisavi** [comment]
* Humans heat up faster based on fire stacks
* Increased the amount of damage you get from being on fire at 480+ bodytemp (8->10)
* Decreased the amount of fire stacks lost over time
* Increased the amount of fire stacks gained from hotspots

TL;DR
A) You will burn out. Eventually.
B) You won't burn out in time to stop your ass from going into crit. Roll.
C) If you're at 20 fire stacks, even if you roll 4 times back to back, you'll still fall just barely into crit. Seek a fire extinguisher. No extinguishers? HONK!
D) A single hot spot is enough to knock you barely into crit, if you do nothing about it. Flammenwerfers are "viable" again.

**Comment by Ergovisavi** [comment]
@Razharas

**Comment by Razharas** [comment]
Looks simple enough, so why not
